pkg/helm/actions/utility.go

```diff
@@ -163,3 +170,46 @@ func getRepositoryConnectionConfig(
 	klog.Errorf("Error listing namespace helm chart repositories: %v \nempty repository list will be used", getClusterRepositoryErr)
 	return v1beta1.ConnectionConfig{}, false, getClusterRepositoryErr
 }
+
+func getSecret(ns string, name string, version int, coreclient corev1client.CoreV1Interface) (Secret, error) {
+	label := fmt.Sprintf("owner=helm,name=%v,version=%v", name, version)
+	timeout := int64(60)
```

Comment:
Our UI has a general timeout after 30 seconds. But what happen after 60 or 30 seconds?

As mentioned above we should create the Secret first and then we might not need a watcher for it.